src/include/kdbdiff.h

```diff
@@ -0,0 +1,43 @@
+#ifndef ELEKTRA_KDBDIFF_H
+#define ELEKTRA_KDBDIFF_H
+
+#include <kdb.h>
+#include <kdbtypes.h>
+
+#ifdef __cplusplus
+namespace ckdb
+{
+extern "C" {
+#endif
+
+typedef struct _ElektraDiff ElektraDiff;
+
+ElektraDiff * elektraDiffCalculate (KeySet * newKeys, KeySet * oldKeys, Key * parentKey);
+
+void elektraDiffDel (ElektraDiff * ksd);
+
+uint16_t elektraDiffIncRef (ElektraDiff * ksd);
+uint16_t elektraDiffDecRef (ElektraDiff * ksd);
+uint16_t elektraDiffGetRef (ElektraDiff * ksd);
+
+bool elektraDiffIsEmpty (const ElektraDiff * ksd);
+
+const Key * elektraDiffGetParentKey (const ElektraDiff * ksd);
+
+KeySet * elektraDiffGetAddedKeys (const ElektraDiff * ksd);
+KeySet * elektraDiffGetRemovedKeys (const ElektraDiff * ksd);
+KeySet * elektraDiffGetModifiedKeys (const ElektraDiff * ksd); // Returns old keys (pre-modification)
```

Comment:
People should read API documentation instead:

```suggestion
KeySet * elektraDiffGetModifiedKeys (const ElektraDiff * ksd);
```